addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -395,10 +398,10 @@ export function useOpenMany2XRecord({
                 isToMany,
             },
             {
-                onClose: () => {
+                onClose: (params) => {
                     resolve();
                     const isNew = !resId;
-                    onClose(isNew);
+                    onClose(Object.assign(params, { isNew }));
```

Comment:
this is an API change, not ideal in 16.0. Maybe pass params as second argument in 16, and keep this version for master?